test/distributed/test_store.py

```diff
@@ -326,6 +326,7 @@ def test_init_pg_and_rpc_with_same_socket(self):
         )
 
         rpc.shutdown()
+        dist.destroy_process_group()
```

Comment:
Nit: these `dist.destroy_process_group()` should probably be in a try: finally construct or something.

Reply:
will address this in next PR to avoid wasting of CI usage. ;-)

Reply:
Hey @Skylion007 , right before I start adding `dist.destroy_process_group()` in a try-catch block, I want to raise some questions that bothers me:
1) Should `destroy_process_group` be put inside a try-catch block? I thought it would be better to have it error out if the calling to `destroy_process_group()` failed. IMO, that being said, should we make `destroy_process_group` idempotent?

2) In the case of testing, we need to make sure that the ProcessGroup created in one test is actually destroyed before entering into the next test. Won't a try-catch block silently hide the error if one occurred in `destroy_process_group`?

Reply:
@Skylion007 I see what you mean. Yes you're right, `destroy_process_group()` should be in a `finally` block.